src/SELF_MappedVector_2D_t.f90

```diff
@@ -120,141 +120,140 @@ subroutine SetInteriorFromEquation_MappedVector2D_t(this,geometry,time)
-  !                                decomp%mpiComm, &
-  !                                decomp%requests(msgCount),iError)
-
-  !                 msgCount = msgCount+1
-  !                 call MPI_ISEND(this%boundary(:,s1,e1,ivar,idir), &
-  !                                (this%interp%N+1), &
-  !                                decomp%mpiPrec, &
-  !                                r2,globalSideId, &
-  !                                decomp%mpiComm, &
-  !                                decomp%requests(msgCount),iError)
-  !               endif
-  !             endif
-
-  !           enddo
-  !         enddo
-  !       enddo
-  !     enddo
-
-  !     decomp%msgCount = msgCount
-  !   endif
-
-  ! endsubroutine MPIExchangeAsync_MappedVector2D_t
-
-  ! subroutine ApplyFlip_MappedVector2D_t(this,decomp,mesh)
-  !   ! Apply side flips to sides where MPI exchanges took place.
-  !   implicit none
-  !   class(MappedVector2D_t),intent(inout) :: this
-  !   type(MPILayer),intent(inout) :: decomp
-  !   type(Mesh2D),intent(in) :: mesh
-  !   ! Local
-  !   integer :: e1,s1,e2,s2
-  !   integer :: i,i2
-  !   integer :: r2,flip,ivar,idir
-  !   integer :: globalSideId
-  !   integer :: bcid
-  !   real(prec) :: extBuff(1:this%interp%N+1)
-
-  !   if(decomp%mpiEnabled) then
-  !     do idir = 1,2
-  !       do ivar = 1,this%nvar
-  !         do e1 = 1,this%nElem
-  !           do s1 = 1,4
-
-  !             e2 = mesh%sideInfo(3,s1,e1) ! Neighbor Element
-  !             s2 = mesh%sideInfo(4,s1,e1)/10
-  !             bcid = mesh%sideInfo(5,s1,e1)
-  !             if(s2 > 0 .or. bcid == 0) then ! Interior Element
-  !               r2 = decomp%elemToRank(e2) ! Neighbor Rank
-
-  !               if(r2 /= decomp%rankId) then
-
-  !                 flip = mesh%sideInfo(4,s1,e1)-s2*10
-  !                 globalSideId = mesh%sideInfo(2,s1,e1)
-
-  !                 ! Need to update extBoundary with flip applied
-  !                 if(flip == 1) then
-
-  !                   do i = 1,this%interp%N+1
-  !                     i2 = this%interp%N+2-i
-  !                     extBuff(i) = this%extBoundary(i2,s1,e1,ivar,idir)
-  !                   enddo
-  !                   do i = 1,this%interp%N+1
-  !                     this%extBoundary(i,s1,e1,ivar,idir) = extBuff(i)
-  !                   enddo
-
-  !                 endif
-  !               endif
-
-  !             endif
-
-  !           enddo
-  !         enddo
-  !       enddo
-  !     enddo
-
-  !   endif
-
-  ! endsubroutine ApplyFlip_MappedVector2D_t
-
-  subroutine SideExchange_MappedVector2D_t(this,mesh,decomp)
+  subroutine MPIExchangeAsync_MappedVector2D_t(this,mesh,resetCount)
+    implicit none
+    class(MappedVector2D_t),intent(inout) :: this
+    type(Mesh2D),intent(inout) :: mesh
+    logical,intent(in) :: resetCount
+    ! Local
+    integer :: e1,s1,e2,s2,ivar,idir
+    integer :: globalSideId,r2,tag
+    integer :: iError
+    integer :: msgCount
+
+    if(resetCount) then
```

Comment:
Same as above, remove resetcount option

Reply:
Resolved in [f24f99d](https://github.com/FluidNumerics/SELF/pull/53/commits/f24f99db9e33d33c61cba24a30e9e8176c9a85ea)